CMakeLists.txt

```diff
@@ -105,6 +105,8 @@ elseif(BASEARCH_S360_FOUND)
     option(WITH_DFLTCC_INFLATE "Build with DFLTCC intrinsics for decompression on IBM Z" OFF)
     option(WITH_CRC32_VX "Build with vectorized CRC32 on IBM Z" ON)
 elseif(BASEARCH_X86_FOUND)
+    option(WITH_AVX512 "Build with AVX512" ON)
+    option(WITH_AVX512_VNNI "Build with AVX512 VNNI extensions" ON)
```

Comment:
You are using `WITH_AVX512VNNI` in other places. I think it is best to use without last underscore. But the option names need to be consistent.

Reply:
I agree with removing the last underscore... Only if it would be list of instruction sets (like AVX512F+AVX512VNNI), the underscore would make sense as it would be essentially replacement for comma.

Things tend to get complicated with AVX512 as it's by definition not single instruction set, but instead a group of individual "feature" sets.